lci_strategic_plugin/src/lci_strategic_plugin.cpp

```diff
@@ -851,6 +860,139 @@ void LCIStrategicPlugin::planWhenDEPARTING(const cav_srvs::PlanManeuversRequest&
       current_state.stamp, intersection_exit_time, crossed_lanelets.front().id(), crossed_lanelets.back().id()));
 }
 
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+void LCIStrategicPlugin::mobilityOperationCb(const cav_msgs::MobilityOperationConstPtr& msg)
+{
+  if (msg->strategy == light_controlled_intersection_strategy_)
+  {
+    ROS_DEBUG_STREAM("Received Schedule message with id: " << msg->m_header.plan_id);
+    approaching_light_controlled_interction_ = true;
+    ROS_DEBUG_STREAM("Approaching light Controlled Intersection: " << approaching_light_controlled_interction_);
+
+    if (msg->m_header.recipient_id == config_.vehicle_id)
+      {
+        street_msg_timestamp_ = msg->m_header.timestamp;
+        ROS_DEBUG_STREAM("street_msg_timestamp_: " << street_msg_timestamp_);
+        parseStrategyParams(msg->strategy_params);
+      }
+    previous_strategy_params_ = msg->strategy_params;
+  }
+  
+}
+
+void LCIStrategicPlugin::BSMCb(const cav_msgs::BSMConstPtr& msg)
+{
+  std::vector<uint8_t> bsm_id_vec = msg->core_data.id;
+  bsm_id_ = BSMHelper::BSMHelper::bsmIDtoString(bsm_id_vec);
+  bsm_msg_count_ = msg->core_data.msg_count;
+  bsm_sec_mark_ = msg->core_data.sec_mark;
+}
+
+void LCIStrategicPlugin::parseStrategyParams(const std::string& strategy_params)
```

Comment:
Its possible that we might want to use this functionality while running other operations like platooning. So not every mobility operation message can be assumed to be for this plugin. Therefore, there needs to be a check on the strategy before this is called to ensure its intended for this plugin. If its not, the message should be ignored

Reply:
Please check 867 its already there.